### PR TITLE
chore: Set permissions for GitHub actions

### PR DESCRIPTION
**Pull request by naveensrinivasan**
Restrict the GitHub token permissions only to the required ones; this way, even if the attackers will succeed in compromising your workflow, they won’t be able to do much.

- Included permissions for the action. https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

[Keeping your GitHub Actions and workflows secure Part 1: Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/)

Signed-off-by: naveen <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -18,8 +18,15 @@ on:
...
 
+permissions:
+  contents: read
+
 jobs:
   analyze:
+    permissions:
+      actions: read  # for github/codeql-action/init to get workflow details
+      contents: read  # for actions/checkout to fetch code
+      security-events: write  # for github/codeql-action/autobuild to send a status report
```
Comment (bsipocz) [inline .github/workflows/codeql-analysis.yml:29]: Based on the log the security events still seem to have read permissions only. Though the rest was indeed set to none. Is this an upstream bug or I misunderstand something?

```
GITHUB_TOKEN Permissions
  Actions: read
  Contents: read
  Metadata: read
  SecurityEvents: read
  ```

**Comment by bsipocz** [comment]
Thank you @naveensrinivasan. The PR is certainly an improvement of the current situation, I'll go ahead and merge, but first would like to understand one minor point.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (naveensrinivasan) [reply .github/workflows/codeql-analysis.yml:29]: AFAIK, the SecurityEvents: write is required for writing events. Thanks!

**Comment by naveensrinivasan** [comment]
> Thank you @naveensrinivasan. The PR is certainly an improvement of the current situation, I'll go ahead and merge, but first would like to understand one minor point.

Thanks! Took care of the conflicts!

**Comment by bsipocz** [comment]
OK, so let's go ahead with this. I suppose that `write` access issue is an upstream bug anyway, as what you did is following the docs, and that it's not being pickedup properly is a GHA issue than an astroquery.

Thank you!

**Pull request merged**